src/abstractdataframe/io.jl

```diff
@@ -1,17 +1,79 @@
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, false),
+         allcols::Bool = !get(io, :limit, false),
+         summary::Bool=true)
+
+Render a data frame to an I/O stream in MIME type `mime`.
+
+# Arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supported MIME types are: `"text/plain"`, `"text/html"`, `"text/latex"`,
+  `"text/csv"`, `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+- `allrows::Bool `: supported only for plain text, HTML and LaTeX MIME types;
+   whether to print all rows, rather than a default number defined in
+   `HTML_NROWS` and `LATEX_NROWS` constants for these MIME types.
+- `allcols::Bool`: supported only for HTML and LaTeX mime types;
+   whether to print all columns, rather than a default number defined in
+   `HTML_NCOLS` and `LATEX_NCOLS` constants for these MIME types.
+- `summary::Bool = true`: supported only for HTML mime type;
+   Whether to print a brief string summary of the data frame.
+
+If mime type is unrecognized `"text/plain"` output is produced.
+
+Keyword arguments that are not used by a specific mime are ignored.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
+Base.show(io::IO, mime::MIME, df::AbstractDataFrame;
+          allrows::Bool = !get(io, :limit, false),
+          allcols::Bool = !get(io, :limit, false),
+          summary::Bool=true) =
+    if mime isa MIME"text/html"
+        _show(io, mime, df, allrows=allrows, allcols=allcols, summary=summary)
+    elseif mime isa MIME"text/latex"
+        _show(io, mime, df, allrows=allrows, allcols=allcols)
+    elseif mime isa MIME"text/csv"
+        printtable(io, df, header = true, separator = ',')
```

Comment:
`allrows` and `allcols` are ignored. Better throw an error if they are `false`.

Reply:
Well I was thinking of `"text/csv"` and `"text/tab-separated-values"` as outputs where we assume `io` has "infinite dimensionality". But I will change it to throw an error.

Reply:
Yeah, but it's kind of weird to change the number of rows/columns to print based on the MIME type rather than based on `io`.